iosMath/render/internal/MTTypesetter.m

```diff
@@ -628,6 +628,7 @@ - (void) createDisplayAtoms:(NSArray*) preprocessed
                     MTMathListDisplay* degree = [MTTypesetter createLineForMathList:rad.degree font:_font style:kMTLineStyleScriptScript];
                     [displayRad setDegree:degree fontMetrics:_styleFont.mathTable];
                 }
+                [displayRad addColor: atom.colorString];
```

Comment:
(This applies to all the addColors below)
Setting textColor applies the color recursively to each element. That may override any color already set by a nested color command. Have it such that addColor only adds it to the current element and not apply it recursively.